src/components/progress-report/workflow/progress-report-workflow.repository.ts

```diff
@@ -146,4 +148,57 @@ export class ProgressReportWorkflowRepository extends DtoRepository(
       changes: { status },
     });
   }
+
+  async getNotificationInfoByReportId(reportId: ID) {
+    const query = this.db
+      .query()
+      .match([
+        node('report', 'ProgressReport', { id: reportId }),
+        relation('in', '', ACTIVE),
+        node('engagement', 'Engagement'),
+        relation('in', '', 'engagement', ACTIVE),
+        node('project', 'Project'),
+        relation('out', '', 'member', ACTIVE),
+        node('member', 'ProjectMember'),
+        relation('out', '', 'user', ACTIVE),
+        node('user', 'User'),
+      ])
+      .optionalMatch([
+        node('user'),
+        relation('out', '', 'email', ACTIVE),
+        node('email', 'EmailAddress'),
+      ])
+      .match([
+        node('member'),
+        relation('out', '', 'roles', ACTIVE),
+        node('role', 'Property'),
+      ])
+      .match([
+        node('engagement'),
+        relation('out', '', ACTIVE),
+        node('language', 'Language'),
+        relation('out', '', ACTIVE),
+        node('languageDisplayName', 'LanguageDisplayName'),
+      ])
+      .match([
+        node('report'),
+        relation('out', '', 'start', ACTIVE),
+        node('start', 'Property'),
+      ])
+      .return<{
+        email?: string;
+        role?: Role[];
+        projectId: ID;
+        languageDisplayName: string;
+        start: CalendarDate;
+      }>([
+        'email.value as email',
+        'role.value as role',
+        'project.id as projectId',
+        'languageDisplayName.value as languageDisplayName',
+        'start.value as start',
+      ])
```

Comment:
This query is confused. It doesn't make sense to return N rows for each user email/role and the also ask for a single project id, language name, and start date. It would be easiest to just split these into two different queries. Or possibly just asking for each service.

Ideally all of this extra info could just come from the `ResourceLoader`
```tsx
resources.load(User, userId);
resources.load(Project, projectId);
resources.load(ProgressReport, reportId);
```
That's much easier than importing each module and calling the service.readOne methods.
However, this doesn't yet work with session for different users, which is why I didn't suggest it.
Maybe this is something I can change with the dataloader library migration.